toolchains/darwin.cmake

```diff
@@ -14,6 +14,14 @@ set(CORE_COMPILE_DEFS PLATFORM_OSX)
 # Build core library with dependencies.
 add_subdirectory(${PROJECT_SOURCE_DIR}/core)
 
+if("${MAPZEN_API_KEY}" STREQUAL "")
```

Comment:
This syntax will check for a definition that was explicitly passed into CMake, which our `Makefile` currently does, but this doesn't automatically work if you're not using the `Makefile` to build (e.g. I use `ninja` almost exclusively on desktop).  We can check for environment variables directly in CMake using `$ENV{MAPZEN_API_KEY}` and then this will work without the `Makefile`.